src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java

```diff
@@ -606,10 +598,23 @@ protected boolean isValidRolesMapping(final RestChannel channel, final String ro
 		}
 
 		if (isReadOnly(rolesMappingConfiguration, role)) {
-			forbidden(channel, "Role '" + role + "' has read-only role-mapping.");
+			methodNotAllowed(channel, "Role '" + role + "' has read-only role-mapping.");
 			return false;
 		}
 		return true;
 	}
 
+	boolean isWriteable(final RestChannel channel, final SecurityDynamicConfiguration<?> configuration, final String name) {
+		if (isHidden(configuration, name)) {
+			notFound(channel, "Resource '"+ name +"' is not available.");
+			return true;
+		}
+
+		if (isReadOnly(configuration, name)) {
+            methodNotAllowed(channel, "Resource '"+ name +"' is read-only.");
+			return true;
+		}
+
+		return false;
```

Comment:
should return `true` when the resource is writable (not hidden or read-only).

Reply:
Will revert back from `isWriteable`. Did not want to unnecessarily negate the value during the check. It was initially `isResourceWriteProtected`

Reply:
`isResourceWriteProtected` implies that resource is *not* writeable. It is better to avoid such implied negation.